vllm/engine/async_llm_engine.py

```diff
@@ -217,10 +217,16 @@ async def step_async(self) -> List[RequestOutput]:
         else:
             output = []
 
-        return self._process_model_outputs(
+        request_outputs = self._process_model_outputs(
             output, scheduler_outputs.scheduled_seq_groups,
             scheduler_outputs.ignored_seq_groups)
 
+        # Log stats.
+        if self.log_stats:
```

Comment:
Nit: i think you can just do the logging before below instead of assigning it to `request_outputs`
```
        return self._process_model_outputs(
            output, scheduler_outputs.scheduled_seq_groups,
            scheduler_outputs.ignored_seq_groups)
```

Reply:
I'm not sure it will work, as `stat_logger.log()` should be called after `seq_group.maybe_set_first_token_time(now)`.
https://github.com/vllm-project/vllm/blob/e8cc7967ff8a6f8432747a9e87ab451d36e1ff57/vllm/engine/llm_engine.py#L467

Reply:
is `first_token_time` used in `stat_logger`? https://github.com/search?q=repo%3Avllm-project%2Fvllm%20first_token_time&type=code

Reply:
My bad, It's not. But, However, I still think that the logging should be after `_process_model_outputs()`. That's because during logging, we calculate the latency between tokens, which I believe should include the processing of the model outputs.
https://github.com/vllm-project/vllm/blob/fe3b5bbc23a99533bc7d4a94ae073828ed025974/vllm/engine/llm_engine.py#L601-L603